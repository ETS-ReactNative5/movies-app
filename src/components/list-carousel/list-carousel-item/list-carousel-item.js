import React from 'react';
import { useNavigation } from '@react-navigation/native';

import {
    StyledImage,
    TitleText,
    ReleaseDateText,
    AverageScore,
    StyledWrapper,
    ScoreWrapper
} from './styled';

import { getTruncatedString } from '../../../utils';

export const ListCarousalItem = ({ imageSrc, title, releaseDate, rating, id }) => {

    const navigation = useNavigation();

    const onPressItem = () => {
        navigation.push("Detail", {
            movieId: id
        })
    }

    return (
        <StyledWrapper
            testID='container-wrapper'
            onPress={onPressItem}
            activeOpacity={0.6}
        >
            <StyledImage
                testID='image-container'
                source={{ uri: `https://image.tmdb.org/t/p/w500/${imageSrc}` }}
            />
            <ScoreWrapper>
                <AverageScore testID='rating-container' >
                    {rating}
                </AverageScore>
            </ScoreWrapper>

            <TitleText testID='title-wrapper'>
                {getTruncatedString(title, 30)}
            </TitleText>
            <ReleaseDateText>
                {releaseDate}
            </ReleaseDateText>
        </StyledWrapper>
    )
};
