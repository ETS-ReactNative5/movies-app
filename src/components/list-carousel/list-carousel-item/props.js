import { string, number } from 'prop-types';

export const propTypes = {
    imageSrc: string,
    title: string,
    rating: number
};

export const defaultProps = {
    imageSrc: '',
    title: '',
    rating: 0
};
