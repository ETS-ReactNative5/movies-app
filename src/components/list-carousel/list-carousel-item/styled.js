import FastImage from 'react-native-fast-image';
import styled from 'styled-components/native';

export const StyledImage = styled(FastImage)`
    resize-mode: cover;
    width: 130px;
    height: 150px;
    border-radius: 15px;
`;

export const TitleText = styled.Text`
    font-weight: bold;
    font-size: 16;
    margin-bottom: 5px;
    width: 130px;
    text-align: center;
    margin-top: 15px;
    color: black;
`;

export const TextWrapper = styled.View``;

export const ReleaseDateText = styled.Text`
    color: black;
`;

export const ScoreWrapper = styled.View`
    position: absolute;
    background: #4a0909;
    top: 145px;
    left: 25px;
    padding: 5px;
    border-radius: 15px;

`;
export const AverageScore = styled.Text`
    font-size: 15;
    color: white;
    font-weight: bold;
`;

export const StyledWrapper = styled.TouchableOpacity` 
    align-items: center;
    padding: 10px;
`;
