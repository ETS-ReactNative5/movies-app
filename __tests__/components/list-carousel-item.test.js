import React from 'react';
import { ListCarousalItem } from '../../src/components/list-carousel/list-carousel-item/list-carousel-item';
import { render, fireEvent } from '@testing-library/react-native';


const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => ({
    useNavigation: () => ({
        push: mockedNavigate,
    }),
    useRoute: jest.fn,
}));

const props = {
    imageSrc: 'image',
    rating: 1234,
    title: 'Title',
    releaseDate: '20 Nov, 2015',
    id: 548263
};

describe('Divider Component', () => {

    it('should match the List Carousel Item component snapshot', () => {
        const { toJSON } = render(
            <ListCarousalItem
                {...props}
            />
        );
        expect(toJSON()).toMatchSnapshot();
    });

    it('should render item title correctly', async () => {
        const { getByTestId } = render(
            <ListCarousalItem
                {...props}
            />);
        expect(getByTestId('title-wrapper').props.children).toBe('Title');
    });

    it('should render item title correctly', async () => {
        const { getByTestId } = render(
            <ListCarousalItem
                {...props}
            />);
        expect(getByTestId('rating-container').props.children).toBe(1234);
    });

    it('should navigate to detail screen', () => {
        const { getByTestId } = render(
            <ListCarousalItem
                {...props}
            />);
        fireEvent(getByTestId('container-wrapper'), 'press');
        expect(mockedNavigate).toHaveBeenCalledWith('Detail', { movieId: 548263 });
    });

});
