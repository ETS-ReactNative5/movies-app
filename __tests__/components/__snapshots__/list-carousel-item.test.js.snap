// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component should match the List Carousel Item component snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
  testID="container-wrapper"
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "borderRadius": 15,
            "height": 150,
            "resizeMode": "cover",
            "width": 130,
          },
        ],
      ]
    }
  >
    <FastImageView
      resizeMode="cover"
      source={
        Object {
          "uri": "https://image.tmdb.org/t/p/w500/image",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="image-container"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#4a0909",
          "borderRadius": 15,
          "left": 25,
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 5,
          "position": "absolute",
          "top": 145,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 15,
            "fontWeight": "bold",
          },
        ]
      }
      testID="rating-container"
    >
      1234
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "marginBottom": 5,
          "marginTop": 15,
          "textAlign": "center",
          "width": 130,
        },
      ]
    }
    testID="title-wrapper"
  >
    Title
  </Text>
  <Text
    style={
      Array [
        Object {},
      ]
    }
  >
    20 Nov, 2015
  </Text>
</View>
`;
